File: WebApiRouter/Models/Pessoafisica.cs
namespace WebApiRouter.Models;

public partial class Pessoafisica
{
    public int Id { get; set; }

    public string Codigo { get; set; } = null!;

    public string Nome { get; set; } = null!;

    public string? Cpf { get; set; }

    public sbyte Ativo { get; set; }

    public string? Telefone { get; set; }

    public string? Senha { get; set; }

    public int TipopessoaId { get; set; }

    public virtual Tipopessoa Tipopessoa { get; set; } = null!;

    public virtual ICollection<Pessoajuridica> Pessoajuridicas { get; set; } = new List<Pessoajuridica>();
}

File: routerapp/lib/models/pessoafisica.dart
class PessoaFisica {
  final int? id;
  final String nome;
  final String cpf;
  final String? telefone;

  PessoaFisica({this.id, required this.nome, required this.cpf, this.telefone});

  factory PessoaFisica.fromJson(Map<String, dynamic> json) {
    return PessoaFisica(
      id: json['id'],
      nome: json['nome'],
      cpf: json['cpf'],
      telefone: json['telefone'],
    );
  }
}
